subprojects/docs/src/docs/release/notes.md

```diff
@@ -158,6 +158,12 @@ This change makes Checkstyle build cache friendly, so that your build does not n
 
 Gradle will now download dependencies from remote repositories in parallel (both metadata and artifacts). It will also make sure that if you build multiple projects in parallel (with `--parallel`) and that 2 projects try to download the same dependency at the same time, that dependency wouldn't be downloaded twice.
 
+### Filtering Java test execution based on test failures
+
+When executing `Test` tasks, you can now pass the [`--failed` command-line option](userguide/java_plugin.html#test_filtering). This option will filter test execution to only the test classes that failed in the last execution.
+
+If there are no previous results, the previous results cannot be read or no tests failed in the last execution, all tests will be executed instead.
```

Comment:
That's another way to do it, but I think it's less convenient.  I was thinking of this sort of work flow:

1. Change code
2. Run `gradle test` and test fails
3. Fix some code run `gradle -t test --failed`
4. In continuous build, fewer and fewer tests will be failing until nothing fails.  Then all the tests will run, which can put you back into the fail-fix-test cycle.

The issue with making the task up-to-date when there are no previously failing tests is that we would need to run the test with the same parameters as before (to get UP-TO-DATE) or filter all tests out and ignore that there were no tests that matched our filter (to get something like ~NO-SOURCE).